backend/app/views/spree/admin/products/_form.html.erb

```diff
@@ -1,89 +1,176 @@
           <%= f.error_message_on :name %>
         <% end %>
       </div>
 
-      <div data-hook="admin_product_form_slug">
+      <div data-hook="admin_product_form_slug" class="col-md-6 col-xs-12">
         <%= f.field_container :slug do %>
           <%= f.label :slug, class: 'required' %>
           <%= f.text_field :slug, :class => 'fullwidth title', :required => true %>
           <%= f.error_message_on :slug %>
         <% end %>
       </div>
 
-      <div data-hook="admin_product_form_description">
+      <div data-hook="admin_product_form_description" class="col-xs-12">
         <%= f.field_container :description do %>
           <%= f.label :description %>
-          <%= f.text_area :description, {:rows => "#{unless @product.has_variants? then '22' else '15' end}", :class => 'fullwidth'} %>
+          <%= f.text_area :description, {:rows => '9', :class => 'fullwidth'} %>
           <%= f.error_message_on :description %>
         <% end %>
       </div>
+
+      <div data-hook="admin_product_form_taxons" class="col-xs-12">
+        <%= f.field_container :taxons do %>
+          <%= f.label :taxon_ids, plural_resource_name(Spree::Taxon) %><br />
+          <%= f.hidden_field :taxon_ids, :value => @product.taxon_ids.join(',') %>
+          <span class="fa fa-caret-down text-field-addon"></span>
+        <% end %>
+      </div>
+
     </div>
 
-    <div class="right col-xs-3" data-hook="admin_product_form_right">
-      <div data-hook="admin_product_form_price">
+    <div class="form-section">
+
+      <div class="col-xs-12">
+        <h4><%= I18n.t(:pricing, scope: [:spree, :product_sections]) %></h4>
+      </div>
+
+      <div data-hook="admin_product_form_price" class="col-lg-3 col-xs-6">
         <%= f.field_container :price do %>
           <%= f.label :price, class: 'required' %>
           <%= f.text_field :price, :value => number_to_currency(@product.price, :unit => ''), :required => true %>
           <%= f.error_message_on :price %>
         <% end %>
       </div>
+      <div class="col-lg-9 hidden-md-down"></div>
+      <div class="col-xs-6 hidden-lg-up"></div>
+
+      <div data-hook="admin_product_form_cost_price" class="col-lg-3 col-xs-6">
+        <%= f.field_container :cost_price do %>
+          <%= f.label :cost_price %>
+          <%= f.text_field :cost_price, :value => number_to_currency(@product.cost_price, :unit => '') %>
+          <%= f.error_message_on :cost_price %>
+        <% end %>
+      </div>
 
-      <% if show_rebuild_vat_checkbox? %>
-        <%= render "spree/admin/shared/rebuild_vat_prices_checkbox", form: f, model_name: "product" %>
-        <div class="clearfix"></div>
-      <% end %>
+      <div data-hook="admin_product_form_cost_currency" class="col-lg-3 col-xs-6">
+        <%= f.field_container :cost_currency do %>
+          <%= f.label :cost_currency %>
+          <%= f.text_field :cost_currency %>
+          <%= f.error_message_on :cost_currency %>
+        <% end %>
+      </div>
```

Comment:
With the new additions by currency widget by @jhawthorn  in #1793, I think that all prices can now occupy a single line.